src/controller/python/chip/clusters/attribute.cpp

```diff
@@ -472,6 +475,22 @@ PyChipError pychip_ReadClient_GetReportingIntervals(ReadClient * pReadClient, ui
     return ToPyChipError(err);
 }
 
+void pychip_ReadClient_GetSubscriptionTimeout(ReadClient * pReadClient, uint32_t * milliSec)
```

Comment:
when you find timeout happens, the existing abort should be enough for the callback/readclient destroy

Reply:
On the contrary, calling `pychip_ReadClient_Abort` after OnError occurs will result in repeated release of memory

Reply:
I think you should not call pychip_ReadClient_Abort when onError happens....

Reply:
if onError happens, you call abort from application to release callback, then callback's onDone would be problematic...

Reply:
According to your idea, the python interface needs to be modified. Now `pychip_ReadClient_Abort` is called in `Shutdown`, which will cause a crash when it is released normally after use on the application side.

Reply:
> if onError happens, you call abort from application to release callback, then callback's onDone would be problematic...in readclient, ondone is called immediately after onerror..

Therefore, add ReadType judgment in `OnDone`. If it is SubscriptionType, hand it over to Abort to release it.